### PR TITLE
New Resource: `azurerm_log_analytics_query_pack_query`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new resource Query Pack Query.

--- PASS: TestAccLogAnalyticsQueryPackQuery_basic (246.11s)
--- PASS: TestAccLogAnalyticsQueryPackQuery_requiresImport (278.19s)
--- PASS: TestAccLogAnalyticsQueryPackQuery_update (342.48s)

### CODE REVIEW

File: website/docs/r/log_analytics_query_pack_query.html.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+resource "azurerm_log_analytics_query_pack" "example" {
+  name                = "example-laqp"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_log_analytics_query_pack_query" "example" {
+  name          = "19952bc3-0bf9-49eb-b713-6b80e7a41847"
+  query_pack_id = azurerm_log_analytics_query_pack.example.id
+  body          = "let newExceptionsTimeRange = 1d;\nlet timeRangeToCheckBefore = 7d;\nexceptions\n| where timestamp < ago(timeRangeToCheckBefore)\n| summarize count() by problemId\n| join kind= rightanti (\nexceptions\n| where timestamp >= ago(newExceptionsTimeRange)\n| extend stack = tostring(details[0].rawStack)\n| summarize count(), dcount(user_AuthenticatedId), min(timestamp), max(timestamp), any(stack) by problemId  \n) on problemId \n| order by  count_ desc\n"
```
Comment (katbyte) [inline website/docs/r/log_analytics_query_pack_query.html.markdown:30]: could we format this betteR?
```suggestion
  body          = <<BODY
    let newExceptionsTimeRange = 1d;
    let timeRangeToCheckBefore = 7d;
    exceptions\n| where timestamp < ago(timeRangeToCheckBefore)\n| summarize count() by problemId\n| join kind= rightanti (\nexceptions\n| where timestamp >= ago(newExceptionsTimeRange)\n| extend stack = tostring(details[0].rawStack)\n| summarize count(), dcount(user_AuthenticatedId), min(timestamp), max(timestamp), any(stack) by problemId  \n) on problemId \n| order by  count_ desc\n
  BODY
```

etc

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
```
Comment (katbyte) [inline internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go:50]: i don't see a standalone complete test?

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+func TestAccLogAnalyticsQueryPackQuery_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_log_analytics_query_pack_query", "test")
+	r := LogAnalyticsQueryPackQueryResource{uuid: uuid.New().String()}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
```
Comment (katbyte) [inline internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go:72]: can we go basic -> complete here

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+resource "azurerm_log_analytics_query_pack" "test" {
+  name                = "acctestlaqp-%[2]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_log_analytics_query_pack_query" "test" {
+  name           = "%[3]s"
+  query_pack_id  = azurerm_log_analytics_query_pack.test.id
+  body           = "let newExceptionsTimeRange = 1d;\nlet timeRangeToCheckBefore = 7d;\nexceptions\n| where timestamp < ago(timeRangeToCheckBefore)\n| summarize count() by problemId\n| join kind= rightanti (\nexceptions\n| where timestamp >= ago(newExceptionsTimeRange)\n| extend stack = tostring(details[0].rawStack)\n| summarize count(), dcount(user_AuthenticatedId), min(timestamp), max(timestamp), any(stack) by problemId  \n) on problemId \n| order by  count_ desc\n"
```
Comment (katbyte) [inline internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go:119]: could we use heredoc syntax here too?

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+resource "azurerm_log_analytics_query_pack" "test" {
+  name                = "acctestlaqp-%[2]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_log_analytics_query_pack_query" "test" {
+  name           = "%[3]s"
+  query_pack_id  = azurerm_log_analytics_query_pack.test.id
+  body           = "let newExceptionsTimeRange = 2d;\nlet timeRangeToCheckBefore = 7d;\nexceptions\n| where timestamp < ago(timeRangeToCheckBefore)\n| summarize count() by problemId\n| join kind= rightanti (\nexceptions\n| where timestamp >= ago(newExceptionsTimeRange)\n| extend stack = tostring(details[0].rawStack)\n| summarize count(), dcount(user_AuthenticatedId), min(timestamp), max(timestamp), any(stack) by problemId  \n) on problemId \n| order by  count_ desc\n"
```
Comment (katbyte) [inline internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go:153]: and here

**Review by katbyte — APPROVED** [review]
LGTM once the comments are address and tests pass ♻️

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go (hunk 5 above)
Comment (neil-yechenwei) [reply internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go:153]: Updated

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go (hunk 4 above)
Comment (neil-yechenwei) [reply internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go]: Updated

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go:50]: Added

File: website/docs/r/log_analytics_query_pack_query.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/log_analytics_query_pack_query.html.markdown:30]: Updated

File: internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go (hunk 4 above)
Comment (neil-yechenwei) [reply internal/services/loganalytics/log_analytics_query_pack_query_resource_test.go]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated PR. Please take another look. Thanks.

**Comment by neil-yechenwei** [comment]
image

**Pull request merged**